.github/workflows/tests.yaml

```
@@ -187,6 +188,21 @@ jobs:
           cache-from: type=gha,scope=cloudserver
           cache-to: type=gha,mode=max,scope=cloudserver
 
+      - name: Build and push cloudserver image test coverage
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          context: .
+          target: testcoverage
+          provenance: false
+          tags: |
+            ghcr.io/${{ github.repository }}:${{ github.sha }}-testcoverage
+          labels: |
+            git.repository=${{ github.repository }}
+            git.commit-sha=${{ github.sha }}
+          cache-from: type=gha,scope=cloudserver
+          cache-to: type=gha,mode=max,scope=cloudserver
```

To be safe and avoid mixing up caches with the production one.
```suggestion
          cache-from: type=gha,scope=cloudserver-test
          cache-to: type=gha,mode=max,scope=cloudserver-test
```

---

We do actually want to mix caches to have faster CI. If the production stage added some dependencies in cache, I'd want to reuse them in the `testcoverage` stage